usaspending_api/download/tests/unit/test_sql_annotations.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+# Imports from your apps
+from usaspending_api.download.filestreaming.download_generation import (
+    apply_annotations_to_sql,
+    _top_level_split,
+    _select_columns,
+)
+
+
+def test_top_level_split_nesting_logic():
+    unnested_string = "sdfoing SPLT noivijf"
+    assert _top_level_split(unnested_string, "SPLT") == ["sdfoing ", " noivijf"]
+    nested_string_1 = "() blah SP pojf"
+    assert _top_level_split(nested_string_1, "SP") == ["() blah ", " pojf"]
+    nested_string_2 = "(SP) blah SP pojf"
+    assert _top_level_split(nested_string_2, "SP") == ["(SP) blah ", " pojf"]
+    nested_string_3 = "(SP) blah SP (pojf SP) SP"
+    assert _top_level_split(nested_string_3, "SP") == ["(SP) blah ", " (pojf SP) SP"]
+    nested_string_4 = "((SP)) blah ()SP (pojf SP) SP"
+    assert _top_level_split(nested_string_4, "SP") == ["((SP)) blah ()", " (pojf SP) SP"]
+
+
+def test_top_level_split_quote_logic():
+    unquoted_string = "oinodsg *** vsdoij"
+    assert _top_level_split(unquoted_string, "***") == ["oinodsg ", " vsdoij"]
+    quoted_string_1 = 'oinodsg "***"*** vsdoij'
+    assert _top_level_split(quoted_string_1, "***") == ['oinodsg "***"', " vsdoij"]
+
+
+def test_top_level_split_unsplittable():
+    unsplittable = "sdoiaghioweg oijfewiwje efoiwnovisper"
+    with pytest.raises(Exception):
+        _top_level_split(unsplittable, "not in this")
+
+
+def test_select_columns_nesting_logic():
+    basic_sql = "SELECT steve,jimmy,bob FROM the bench"
+    assert _select_columns(basic_sql) == [" steve", "jimmy", "bob "]
```

Comment:
Shouldn't the extra whitespace be removed from the column names?

Reply:
In an effort to maintain Single Responsibly Principle, `_select_columns` only separates out the text for each column, instead of doing any formatting for the column. Perhaps a deceptive functionality, but I'd still rather rename the helper than conjoin different parts of logic into one function.

Reply:
If the column names don't start with leading or trailing whitespace then one could make the argument that `_select_columns()` isn't actually returning the list of column names

Reply:
That's true. Furthermore, I can't think of a use case where you WANT the whitespace when using that function.